### PR TITLE
SLM-270: upgrade cypress

### PR DESCRIPTION
**Pull request by tpmcgowan**
Upgrade Cypress to version 12 and fix integration and smoke tests to work with this. Also upgrade wiremock to latest version in Docker and CircleCI config.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "scripts": {
     "prepare": "husky install",
     "copy-views": "cp -R server/views dist/server/",
-    "compile-sass": "sass --no-source-map --load-path=node_modules/govuk-frontend --load-path=node_modules/@ministryofjustice/frontend --load-path=. ./assets/sass/application.sass:./assets/stylesheets/application.css ./assets/sass/application-ie8.sass:./assets/stylesheets/application-ie8.css --style compressed",
+    "compile-sass": "sass --quiet-deps --no-source-map --load-path=node_modules/govuk-frontend --load-path=node_modules/@ministryofjustice/frontend --load-path=. ./assets/sass/application.sass:./assets/stylesheets/application.css ./assets/sass/application-ie8.sass:./assets/stylesheets/application-ie8.css --style compressed",
```
Comment (tpmcgowan) [inline package.json:10]: Not related to this Cypress change, but added `--quiet-deps` to suppress noisy Sass compilation warnings for dependencies

**Comment by tpmcgowan** [comment]
@dhirajshettymoj I've got Cypress upgraded and working fine for the integration tests. I _think_ the smoke tests are fixed too: they run fine locally but to be sure we'll just have to merge this in and see how they go in the CircleCI pipeline 🤞

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+16
```
Comment (tpmcgowan) [inline .nvmrc:1]: Adding this so that NVM can pick up the correct node version for this project

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**